buildenv/jenkins/omrMirror

```diff
@@ -54,7 +55,22 @@ pipeline {
                                         returnStdout: true
                                     ).trim()
                                 currentBuild.description = "${LAST_COMMIT}"
+
+                                current_sha = sh (
+                                        script: 'git rev-parse --short HEAD',
+                                        returnStdout: true
+                                    ).trim()
+                            }
+
+                            // if there were changes, launch an acceptance build
+                            step([$class: 'CopyArtifact', optional: true, fingerprintArtifacts: true, projectName: 'Mirror-OMR-to-OpenJ9-OMR'])
+                            def default_sha = [omr_sha: 'Default']
+                            def previous = readProperties defaults: default_sha, file: "${ARCHIVE_FILE}"
+                            if ( previous.omr_sha != current_sha ) {
+                                build job: 'Pipeline-OMR-Acceptance', propagate: false, wait: false
                             }
+                            writeFile file: "${ARCHIVE_FILE}", text: "omr_sha=${current_sha}"
```

Comment:
Shouldn't this be inside the if block?  As I read it now (and please correct me if I'm wrong), it will always write the file.

Reply:
becasue the job every 15 minutes, and becasue the `CopyArtifact` step pulls the artifact from the last successful build, we need to archive the artifact every build.
Although, I suppose if we don't have no new content there is no need to overwrite the file with the same SHA. The only other case would be if we failed to fetch the previous artifact, then we would get the default values, in which case we would have "new content" and we would write the file.
So I think you are correct we should be safe to move it inside the `if`

Reply:
Updated commit